seed/ts-sdk/websocket/websocket-no-serde/src/api/resources/realtime/client/Socket.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * This file was auto-generated by Fern from our API Definition.
+ */
+
+import * as core from "../../../../core";
+import * as SeedWebsocket from "../../../index";
+import { fromJson, toJson } from "../../../../core/json";
+
+export declare namespace RealtimeSocket {
+    export interface Args {
+        socket: core.ReconnectingWebSocket;
+    }
+
+    export type Response = SeedWebsocket.ReceiveEvent | SeedWebsocket.ReceiveEvent2 | SeedWebsocket.ReceiveEvent3;
+    type EventHandlers = {
+        open?: () => void;
+        message?: (message: Response) => void;
+        close?: (event: core.CloseEvent) => void;
+        error?: (error: Error) => void;
+    };
+}
+
+export class RealtimeSocket {
+    public readonly socket: core.ReconnectingWebSocket;
+    protected readonly eventHandlers: RealtimeSocket.EventHandlers = {};
+    private handleOpen: () => void = () => {
+        this.eventHandlers.open?.();
+    };
+    private handleMessage: (event: { data: string }) => void = (event) => {
+        const data = fromJson(event.data);
+
+        this.eventHandlers.message?.(data as RealtimeSocket.Response);
+    };
+    private handleClose: (event: core.CloseEvent) => void = (event) => {
+        this.eventHandlers.close?.(event);
+    };
+    private handleError: (event: core.ErrorEvent) => void = (event) => {
+        const message = event.message;
+        this.eventHandlers.error?.(new Error(message));
+    };
+
+    constructor(args: RealtimeSocket.Args) {
+        this.socket = args.socket;
+        this.socket.addEventListener("open", this.handleOpen);
+        this.socket.addEventListener("message", this.handleMessage);
+        this.socket.addEventListener("close", this.handleClose);
+        this.socket.addEventListener("error", this.handleError);
+    }
+
+    /** The current state of the connection; this is one of the readyState constants. */
+    get readyState(): number {
+        return this.socket.readyState;
+    }
+
+    /**
+     * @param event - The event to attach to.
+     * @param callback - The callback to run when the event is triggered.
+     */
+    public on<T extends keyof RealtimeSocket.EventHandlers>(event: T, callback: RealtimeSocket.EventHandlers[T]) {
```

Comment:
nit it would be great to add some docstrings about how to use this: 
`socket.on["open"]...`